Bence

kornisb opened the pull request.

BotondSoltesz commented on pancake-service/pom.xml:
```diff
@@ -0,0 +1,42 @@
+			<groupId>org.springframework</groupId>
+			<artifactId>spring-context</artifactId>
+		</dependency>
+		<!-- Testing -->
```
I suggest to move Test dependencies after Non-test dependencies.

BotondSoltesz commented on pancake-service/pom.xml:
```diff
@@ -0,0 +1,42 @@
+		</dependency>
+		<dependency>
+			<groupId>io.github.pancake.persistence</groupId>
+			<artifactId>io-github-pancake-persistence-impl</artifactId>
```
Please consider to use runtime scope.

BotondSoltesz commented on pancake-service/src/test/java/io/github/pancake/PancakeServiceTest.java:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    @Test
+    public void testServiceShouldReturnPancakeTypes() {
```
Please rename to testGetAccessiblePancakeTypes...

kicsikrumpli replied:
Since we are renaming stuff, probably available might be a better phrase than accessible

BotondSoltesz commented on pancake-service/src/main/java/io/github/pancake/service/PancakeService.java:
```diff
@@ -0,0 +1,24 @@
+     * @return list of {@link Pancake}.
+     */
+    public List<Pancake> getAccessiblePancakeTypes() {
+        return Arrays.asList(Pancake.values());
```
I suggest not to create _List of Pancake_ every call of getAccessiblePancakeTypes() method because it seems wasting resources.

BotondSoltesz commented on pancake-service/src/test/java/io/github/pancake/PancakeServiceTest.java:
```diff
@@ -0,0 +1,40 @@
+        List<Pancake> pancakeTypesFromService = underTest.getAccessiblePancakeTypes();
+
+        // THEN
+        Assert.assertTrue(pancakeTypes.size() == pancakeTypesFromService.size());
```
If Assert.assertEquals can be used please use it instead of using Assert.assertTrue with logical expression.

BotondSoltesz commented on pancake-webapp/src/main/java/io/github/pancake/app/ServletDemo.java:
```diff
@@ -0,0 +1,51 @@
+        out.println("<html>");
+        out.println("<body>");
+        for (Pancake pancakeType : pancakeTypes) {
+            out.println(String.format("<p>%s</p>", pancakeType.toString()));
```
I suggest to use _name()_ method instead of _toString()_

kicsikrumpli replied:
+1 on this - toString should only be used for describing an object in a log message.

BotondSoltesz commented on pancake-webapp/src/test/java/io/github/pancake/app/ServletDemoTest.java:
```diff
@@ -35,6 +41,7 @@ public void setUp() {
     public void testDoGetShouldPrintStaticContentWhenInvoked() throws Exception {
         // GIVEN
         Mockito.when(mockResponse.getWriter()).thenReturn(mockPrintWriter);
+        Mockito.when(mockPancakeService.getAccessiblePancakeTypes()).thenReturn(Collections.emptyList());
```
Please consider to use static import.

BotondSoltesz commented on pom.xml:
```diff
@@ -1,55 +1,99 @@
-	<version>1.0-SNAPSHOT</version>
-	<packaging>war</packaging>
+	<artifactId>pancake</artifactId>
+	<version>1.0</version>
```
Version: 1.0-SNAPSHOT

kicsikrumpli commented on pancake-service/src/test/java/io/github/pancake/PancakeServiceTest.java:
```diff
@@ -0,0 +1,40 @@
+    public void testServiceShouldReturnPancakeTypes() {
+        // GIVEN
+        List<Pancake> pancakeTypes = new ArrayList<>(
+                Arrays.asList(Pancake.COCOA, Pancake.COTTAGE_CHEES, Pancake.CINNAMON, Pancake.NUTELLA));
```
I am not sure I like how this test depends on the Pancake enum. Get available pancakes should return all available pancakes. So what we should test is if all pancakes are returned instead of checking, if COCOA, COTTAGE_CHEESE, CINNAMON and NUTELLA are returned.

kornisb commented:
Thanks for the comments, I fixed most of the comments, and refactored the code in the meanwhile.
I closed the pull request so I can create a new one.
